src/main/java/de/tum/in/www1/artemis/web/rest/ResultResource.java

```diff
@@ -276,4 +280,39 @@ public ResponseEntity<Result> createResultForExternalSubmission(@PathVariable Lo
         return ResponseEntity.created(new URI("/api/results/" + savedResult.getId()))
                 .headers(HeaderUtil.createEntityCreationAlert(applicationName, true, ENTITY_NAME, savedResult.getId().toString())).body(savedResult);
     }
+
+    /**
+     * GET exercises/:exerciseId/feedback-details : Retrieves all negative feedback details and the latest result IDs for a given exercise.
+     *
+     * @param exerciseId The ID of the exercise for which feedback details and result IDs should be retrieved.
+     * @return A ResponseEntity containing a list of all feedback details and the corresponding result IDs for the exercise.
+     */
+    @GetMapping("exercises/{exerciseId}/feedback-details")
+    @EnforceAtLeastEditorInExercise
+    public ResponseEntity<FeedbackDetailsWithResultIdsDTO> getAllFeedbackDetailsForExercise(@PathVariable Long exerciseId) {
+        log.debug("REST request to get all Feedback details for Exercise {}", exerciseId);
+
+        List<StudentParticipation> participations = studentParticipationRepository
+                .findByExerciseIdWithLatestAutomaticResultAndFeedbacksAndTestCasesWithoutIndividualDueDate(exerciseId);
+        removeSubmissionAndExerciseData(participations);
+
+        List<Long> resultIds = new ArrayList<>();
+
+        List<FeedbackDetailDTO> allFeedbackDetails = new ArrayList<>(participations.stream().filter(participation -> !participation.isPracticeMode())
+                .flatMap(participation -> participation.getResults().stream()).peek(result -> resultIds.add(result.getId())).flatMap(result -> result.getFeedbacks().stream())
+                .filter(feedback -> Boolean.FALSE.equals(feedback.isPositive()))
```

Comment:
So this means you consider a feedback positive in case `isPositive` is `null` for whatever reason? 🤔